src/services/mongoose/ln-payments.ts

```diff
@@ -0,0 +1,60 @@
+import { toSats } from "@domain/bitcoin"
+import {
+  CouldNotFindLnPaymentFromHashError,
+  UnknownRepositoryError,
+} from "@domain/errors"
+import { LnPayment } from "@services/lnd/schema"
+
+export const LnPaymentsRepository = (): ILnPaymentsRepository => {
+  const findByPaymentHash = async (
+    paymentHash: PaymentHash,
+  ): Promise<LnPayment | RepositoryError> => {
+    try {
+      const result = await LnPayment.findOne({ paymentHash })
+      if (!result) {
+        return new CouldNotFindLnPaymentFromHashError(paymentHash)
+      }
+      return lnPaymentFromRaw(result)
+    } catch (err) {
+      return new UnknownRepositoryError(err)
+    }
+  }
+
+  const update = async (
+    payment: LnPaymentLookup,
+  ): Promise<LnPayment | RepositoryError> => {
+    try {
+      const result = await LnPayment.findOneAndUpdate(
+        { paymentHash: payment.paymentHash },
+        payment,
+        { upsert: true, new: true },
```

Comment:
this does not insert default values, if in this case we want to use them then we should use:
```
{ upsert: true, new: true, setDefaultsOnInsert: true },
```

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/665/commits/bd9233cf74c4e0b00758585e4e40a118c1b61cc8